include/eve/detail/function/simd/riscv/combine.hpp

```diff
@@ -33,10 +30,11 @@ requires rvv_abi<abi_t<T, N>>
     auto           wider_l        = simd_cast(l, as<that_t> {});
     auto           wider_h        = simd_cast(h, as<that_t> {});
     constexpr auto shift_size     = N::value;
+    constexpr size_t combined_vl    = N::combined_type::value;
     that_t         wider_h_placed = __riscv_vslideup(wider_h, wider_h, shift_size, combined_vl);
-    auto           mask_all_ones  = rvv_true<T, N>();
-    auto           wider_mask     = simd_cast(mask_all_ones, as<logical<that_t>> {});
-    return if_else(wider_mask, wider_l, wider_h_placed);
+    // TODO: can be optimized when simd_cast will support conversions wide<->logical
```

Comment:
I'd suggest to do it here, since you are working on simd cast.

Reply:
It seems this patch will be too big.
Moreover, I'd like to try out another approach to mask initialization. My previous idea of  using separate functions in `rvv_utils` to generate `logical` seems a little bit overcomplicated.

I think this could be generalized with a lambda as a template parameter to create a correct `logical`, though I'm still figuring out how to make it pretty enough.

I'd like for this patch to focus on enabling basic support for `simd_cast`, like it is used in other architectures. This will unblock other prepared patches for RISC-V, while I'm preparing support for casts between `logical` and `wide` for RISC-V.

If you insist, I'll add support with separate functions in `rvv_utils`, like it was done before.